src/panels/calendar/dialog-calendar-event-editor.ts

```diff
@@ -73,7 +73,15 @@ class DialogCalendarEventEditor extends LitElement {
       }
     } else {
       this._allDay = false;
-      this._dtstart = startOfHour(new Date());
+      // If we have been provided a selected date (e.g. based on the currently displayed
+      // day in a calendar view), use that as the starting value.
+      if (params.selectedDate) {
+        const startingDate = params.selectedDate;
+        startingDate.setHours(startOfHour(new Date()).getHours());
+        this._dtstart = startingDate;
```

Comment:
Is this equivalent?
```
this._dtstart = startOfHour(params.selectedDate);
```

I'm wondering if this can be combined into a one-liner that checks the selected date or a default value of `new Date`

Reply:
Yes, indeed. My version was basically a relic from my initial tests and not optimized enough. Thanks for the pointer.